src/java.base/share/classes/jdk/internal/classfile/impl/CodeImpl.java

```diff
@@ -291,11 +291,12 @@ else if (frameType < 128) {
             }
             else {
                 switch (frameType) {
-                    case 246 -> {
+                    case StackMapDecoder.EARLY_LARVAL -> {
                         int numberOfUnsetFields = classReader.readU2(p + 1);
                         p += 3;
                         p += 2 * numberOfUnsetFields;
-                        continue; // do not move bci/create label
+                        i--; // one more enclosed frame
+                        continue;
                     }
                     case 247 -> {
```

Comment:
Although not related to this PR, why is it that only StackMapDecoder.EARLY_LARVAL uses constants, but the others use magic numbers?

Reply:
We periodically merge from mainline JDK. These are in mainline; we don't want to create redundant merge conflicts. There is a mainline bug entry for this, feel free to contribute: https://bugs.openjdk.org/browse/JDK-8354871

I think such a contribution will be more welcomed by other reviewers compared to domain-specific optimizations.